corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py

```diff
@@ -23,6 +22,145 @@ def get_case_payload(restore_state):
 chunk_size = 1000
 
 
+class CleanOwnerSyncPayload(object):
+    def __init__(self, case_ids_to_sync, restore_state):
+        self.restore_state = restore_state
+        self.case_accessor = CaseAccessors(self.restore_state.domain)
+        self.response = self.restore_state.restore_class()
+
+        self.case_ids_to_sync = case_ids_to_sync
+        self.all_maybe_syncing = copy(case_ids_to_sync)
+        self.checked_cases = set()
+        self.child_indices = defaultdict(set)
+        self.extension_indices = defaultdict(set)
+        self.all_dependencies_syncing = set()
+        self.closed_cases = set()
+        self.potential_elements_to_sync = {}
+
+    def get_payload(self):
+        while self.case_ids_to_sync:
+            self.process_case_batch(self._get_next_case_batch())
+        self.update_index_trees()
+        self.update_case_ids_on_phone()
+        self.move_no_longer_owned_cases_to_dependent_list_if_necessary()
+        irrelevant_cases = self.purge_and_get_irrelevant_cases()
+        self.compile_response(irrelevant_cases)
+
+        return self.response
+
+    def _get_next_case_batch(self):
+        ids = pop_ids(self.case_ids_to_sync, chunk_size)
+        # TODO: see if we can avoid wrapping - serialization depends on it heavily for now
```

Comment:
Don't think this is really relevant anymore now that we have SQL and Couch cases. If we didn't wrap couch cases we'd have to convert SQL cases to JSON which is probably as much effort.